packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1903,7 +1935,15 @@ export class LinuxBuildImage implements IBuildImage {
   }
 
   public validate(_env: BuildEnvironment): string[] {
-    return [];
+    const ret = [];
+
+    const lambdaComputeTypes = Object.values(ComputeType).filter(value => value.startsWith('BUILD_LAMBDA'));
+    if (_env.computeType && lambdaComputeTypes.includes(_env.computeType)) {
+      ret.push(`x86-64 images only support ComputeTypes between '${ComputeType.SMALL}' and '${ComputeType.X2_LARGE}' - ` +
+               `'${_env.computeType}' was given`);
+    }
+
+    return ret;
```

Comment:
```suggestion
    const ret = [];

    const lambdaComputeTypes = Object.values(ComputeType).filter(value => value.startsWith('BUILD_LAMBDA'));
    if (this.isLambdaComputeType(env.computeType)) {
      ret.push('x86-64 images do not support Lambda compute mode');
    }

    return ret;
```
It's better to declare a separate function to check if the compute type is a Lambda one and re-use it.
```
private isLambdaComputeType(computeType?: string): boolean {
  if (!computeType) return false;
  return [
    ComputeType.LAMBDA_1GB,
    ComputeType.LAMBDA_2GB,
    ...
  ].includes(computeType);
}

Reply:
added implementation as a function instead of a method so that it can be used in other files.